app/src/main/java/com/imashnake/animite/features/MainActivity.kt

```diff
@@ -39,65 +44,72 @@ class MainActivity : ComponentActivity() {
 
-                DestinationsNavHost(
-                    navGraph = NavGraphs.root,
-                    modifier = Modifier
-                        .align(Alignment.TopCenter)
-                        .background(Backdrop),
-                    navController = navController,
-                    engine = navHostEngine
-                )
+                    DestinationsNavHost(
+                        navGraph = NavGraphs.root,
+                        modifier = Modifier
+                            .align(Alignment.TopCenter)
+                            .background(MaterialTheme.colorScheme.background),
+                        navController = navController,
+                        engine = navHostEngine
+                    )
 
-                SearchBar(
-                    modifier = Modifier
-                        .align(Alignment.BottomEnd)
-                        .padding(bottom = 80.dp, start = 24.dp, end = 24.dp)
-                        .navigationBarsPadding()
-                        .padding(bottom = 24.dp),
-                    navController = navController
-                )
+                    val bottomPadding: Dp by animateDpAsState(
+                        if (
+                            NavigationBarPaths.values().map { it.direction }.any {
+                                navController.appCurrentDestinationAsState().value == it
+                            }
+                        ) bottomNavigationBarSize else 0.dp
+                    )
 
-                AnimatedVisibility(
-                    visible = NavigationBarPaths.values().map { it.direction }.any {
-                        navController.appCurrentDestinationAsState().value == it
-                    },
-                    modifier = Modifier
-                        .align(Alignment.BottomCenter)
-                        .height(
-                            80.dp + WindowInsets
-                                .navigationBars
-                                .asPaddingValues()
-                                .calculateBottomPadding()
-                        ),
-                    enter = slideInVertically { it },
-                    exit = slideOutVertically { it }
-                ) {
-                    NavigationBar(
-                        navController = navController,
+                    SearchBar(
+                        modifier = Modifier
+                            .align(Alignment.BottomEnd)
+                            .padding(
+                                start = 24.dp,
+                                end = 24.dp,
+                                bottom = bottomPadding
+                            )
+                            .navigationBarsPadding()
+                            .padding(bottom = 24.dp),
+                        navController = navController
+                    )
+
+                    AnimatedVisibility(
+                        visible = NavigationBarPaths.values().map { it.direction }.any {
```

Comment:
This will complete one full iteration again, then potentially search the whole list. Consider using a single operator for performance, `any { it.destination == navController.appCurrentDestinationAsState().value }` should work fine